### ISSUE TITLE
(typescript) optional property not highlighted correctly

### ISSUE THREAD

**Issue by unional** [highlightjs/highlight.js#3613]
**Describe the issue**
 optional property (`prop?: number`) not highlighted correctly.

**Which language seems to have the issue?**
TypeScript

**Are you using `highlight` or `highlightAuto`?**
highlight? Using highlightjs through revealjs VSCode extension.

**Sample Code to Reproduce**

```ts
type Options = {
  srcdir: string,
  outdir: string,
  minify?: boolean
}
```

Result (using Github Dark):
![image](https://user-images.githubusercontent.com/3254987/189471981-eab01afc-e85d-4efb-83c4-060e418d082c.png)

**Expected behavior**

As seen above, rendered by Github

**Comment by joshgoebel** [comment]
In TS can you write: `prop ?: number`?

**Comment by unional** [comment]
Yes. That is a valid syntax.

**Comment by Tokimon** [comment]
I have the same problem even almost a year later. It might work to insert the space, but it is inconvenient apart from being an incorrect parsing of the properties.
Any prospects of this being corrected?

**Comment by iSuslov** [comment]
Bumping it up. 


> In TS can you write: `prop ?: number` ?

@joshgoebel yes, it is used all over the place in typescript. More often without space in between key name and a question mark. 

[Example from the typescript repo](https://github.com/microsoft/TypeScript/blob/e154d47851797a58dc32e8d1049ae1e68a210023/src/server/protocol.ts#L277).

**Comment by Dxuian** [comment]
@joshgoebel  i believe i've managed to fix this please review #4114
